src/main/kotlin/com/exactpro/th2/codec/api/IPipelineCodec.kt

```diff
@@ -19,10 +19,12 @@ package com.exactpro.th2.codec.api
 import com.exactpro.th2.common.grpc.MessageGroup
 
 interface IPipelineCodec : AutoCloseable {
-    fun encode(messageGroup: MessageGroup): MessageGroup
+    @Deprecated("Deprecated due new functionality", ReplaceWith("encode(MessageGroup, IReportingContext)"))
```

Comment:
This replacement won't work because it should contain the correct parameter names but it does not